### DISCUSSION TITLE
Radio Signal Range Degradation

### DISCUSSION THREAD

**Discussion by SoobGorl** [FakeFishGames/Barotrauma#4754] [Ideas and suggestions]
There is a pretty large gap between players that have mic, and don't, extending far past reaction time. With text-chat, you only have to be a few dozens of meters outside of a sub before your messages become incomprehensible, whilst players that use voice chat suddenly cut out after being crystal clear when traveling far from the sub by a few hundred meters. To balance this problem a bit, it would be interesting if something like a compressor/bit crusher was applied to players using voice chat the further you, or they, move away from you. An example of this can be found below (sound on):

(Numbers may be widely inaccurate, as I'm pulling information from past experiences on the voice-radio range, but the basic idea is the farther you go, the less easy you are to  be heard, etc)

https://user-images.githubusercontent.com/72813628/104629685-cca64100-5667-11eb-9663-81ea7b112f13.mp4

Edit: Could also be applied to EMP enemies that temporarily "blind" radio communications, but that could be more annoying than fun.

**Comment by Ek-Videogames** [comment]
+1
( Being able to hear voicechat hundreds of meters away past text chat is presumably a bug either way )

**Reply by SnippitySnap1** [reply]
Hearing that kind of static out of nowhere almost would scare me if it is nearly as loud as that

**Comment by Saevon** [comment]
First off the static should be a bit less in your face. Let's keep playability in mind for the sake of my ears.

Secondly how would this work on "wifi" channels?  getting ½ the signals through could be worse than none at all. 
 * Add a `signal degrades` option (with help hover `Use for text-communication which would replace some letters with * as signal degrades.  Disable for data signals so they stop working suddenly, predictably affecting circuits`)

**Comment by NilanthAnimosus** [comment]
I have never liked the discrepancy in voice and chat range, I primarily use text communications. And the distance text is garbled from voice is very disproportionate, offering essentially a bonus to mic users.

As a headphone user however, I don't want large static-filled ear-destroying sound spikes as much as the next person, Perhaps there is a way to both distort and clip the volume of the audio to distance, while maintaining a background noise that is tolerable that does not get distorted merged into it, but enough to prevent merely jacking up the volume to make it out easily. If tweaked well enough it could make understanding the noise is an out of range voice radio then, but not from whom, or quite make out what is said if too far.

As for wifi channels/components, I think they are fine as-is. I'm not strictly after the buffing of text communications (Unless that simpler approach is the only real way about it). they can go far enough as it is at most only needing a buff for same-sub communication distances for when a submarine is just too long.
